controllers/progressiverollout_controller_test.go

```diff
@@ -22,6 +25,26 @@ const (
 	interval = time.Millisecond * 10
 )
 
+type mockArgoCDAppClientCounter struct {
+	appsSynced []string
+	m          sync.Mutex
+}
+
+func (c *mockArgoCDAppClientCounter) Sync(ctx context.Context, in *applicationpkg.ApplicationSyncRequest, opts ...grpc.CallOption) (*argov1alpha1.Application, error) {
+	c.m.Lock()
+	c.appsSynced = append(c.appsSynced, *in.Name)
+	defer c.m.Unlock()
```

Comment:
as a convention, if you're deferring the unlock, it's good to keep it next to the `Lock()`